cpp/mcap/include/mcap/reader.hpp

```diff
@@ -510,6 +556,39 @@ struct TypedRecordReader {
   bool parsingChunk_;
 };
 
+struct IndexedMessageReader {
```

Comment:
maybe this could move to a separate file? forward-declare and place in the implementation?